provider/identityprovider/identity_provider_resource.go

```diff
@@ -581,7 +584,114 @@ func (r *IdentityProviderResource) Read(ctx context.Context, request resource.Re
 
 func (r *IdentityProviderResource) Update(ctx context.Context, request resource.UpdateRequest,
 	response *resource.UpdateResponse) {
-	response.Diagnostics.AddError("IDP Update not supported.", "This RHCS provider version does not support updating an existing IDP")
+	// Get the state:
+	state := &IdentityProviderState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if state.HTPasswd == nil {
+		response.Diagnostics.AddError("IDP Update not supported for non-HTPasswd IDPs.",
+			"This RHCS provider version does not support updating an existing IDP")
+		return
+	}
+
+	// Get the plan:
+	plan := &IdentityProviderState{}
+	diags = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	plan.ID = state.ID
+
+	resource := r.collection.Cluster(state.Cluster.ValueString()).IdentityProviders().
+		IdentityProvider(state.ID.ValueString())
+
+	if !reflect.DeepEqual(state.HTPasswd.Users, plan.HTPasswd.Users) {
+		stateUserMap, err := HTPasswdUserListToStringMaps(ctx, state.HTPasswd.Users, resource)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't get identity provider user ID",
+				fmt.Sprintf(
+					"Can't get identity provider user ID with identifier for "+
+						"cluster '%s': %v",
+					state.Cluster.ValueString(), err,
+				),
+			)
+			return
+		}
+		planUserMap, err := HTPasswdUserListToStringMaps(ctx, plan.HTPasswd.Users, resource)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't get identity provider user ID",
+				fmt.Sprintf(
+					"Can't get identity provider user ID with identifier for "+
+						"cluster '%s': %v",
+					state.Cluster.ValueString(), err,
+				),
+			)
+			return
+		}
+
+		var removedUsers []string
+
+		for user, htpasswdUser := range stateUserMap {
+			if planUserMap[user].Username == "" { // Not in plan, delete
+				err = htpasswd.DeleteUser(ctx, htpasswdUser.Id, resource)
```

Comment:
I still think we need to setup a separate package for all of these, there's just a lot of things in here which are only for the htpsswd idp

Reply:
Oh you did it, but internally I see.

I wonder if we can capture the whole patch of htpsswd into a function then. Just because when we get to the other idps it will be a lot of code

Reply:
Just wrapping into a function call, like patchHtpasswordUserList or something, so from the update flow it is clearer what is happening. Eventually we'll have the other idps too, so code will get confusing

Reply:
Tried moving all the logic to this pkg, including even moving the `htpasswd.go` file, but I cannot get around having an import cycle due to the state/plan vars. Any suggestions on this?
The API funcs are fine due to them not using the state nor plan, but moving the other logic over creates this cycle. The `IdenittyProviderState` struct is part of a file shared by all identity providers.

Technically, I could move the resource file into its own package, but im not sure if that's best.

Reply:
i think tomorrow AM im gonna try to move all the files which import the resource file into their own dirs, not sure if thatll work but just thought of it now

Reply:
@hunterkepley I don't understand the issue here
What prevents from moving all this code to dedicated update function inside the htpasswd file?
Everything added here from line 613 is specific for htpasswd, no?

Reply:
@nirarg There is an import cycle preventing me from making the binary, it is due to the `IdentityProviderState` and resource structs. I can move the state file to its own package, but it's shared by all identityproviders. I will do this if you think that's alright, but i cannot move this code with the current file structure